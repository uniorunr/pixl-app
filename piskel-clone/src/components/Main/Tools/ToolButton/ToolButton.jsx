import React, { Component } from 'react';
import './ToolButton.scss';
import PropTypes from 'prop-types';

class ToolButton extends Component {
  makeActive = () => {
    const { updateCurrentTool, id } = this.props;
    updateCurrentTool(id);
  };

  render() {
    const { buttonClass, id, iconClass } = this.props;
    return (
      <button
        className={buttonClass}
        id={id}
        data-listener="true"
        type="button"
        onClick={this.makeActive}
      >
        <i className={iconClass} />
      </button>
    );
  }
}

ToolButton.propTypes = {
  buttonClass: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  iconClass: PropTypes.string.isRequired,
  updateCurrentTool: PropTypes.func.isRequired,
};

export default ToolButton;
