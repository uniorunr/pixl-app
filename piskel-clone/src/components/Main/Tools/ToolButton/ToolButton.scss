.tools__button {
  width: 35px;
  height: 35px;
  position: relative;
  top: 0;
  margin-bottom: 15px;
  background-color: white;
  border: 2px solid black;
  box-shadow: 2px 2px 0 black;
  cursor: pointer;
  outline: none;

  .fas {
    display: flex;
    justify-content: center;
    font-size: 14px;
  }
}

#rectangle.tools__button, #circle.tools__button {
  font-size: 16px;
}

.tools__button:active {
  top: 2px;
  box-shadow: 0 0 0 black;
}

.tools_state_active {
  border: 3px dashed black;
}
