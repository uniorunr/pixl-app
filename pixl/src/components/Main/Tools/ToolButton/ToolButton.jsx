import React, { Component } from 'react';
import './ToolButton.scss';
import PropTypes from 'prop-types';

class ToolButton extends Component {
  makeActive = () => {
    const { updateCurrentTool, id } = this.props;
    updateCurrentTool(id);
  };

  render() {
    const {
      buttonClass, id, iconClass, shortcut,
    } = this.props;
    return (
      <div className="tools__container">
        <button
          className={buttonClass}
          id={id}
          data-listener="true"
          type="button"
          onClick={this.makeActive}
        >
          <i className={iconClass} />
        </button>
        <div className="tool__tooltip">{`${id} (${shortcut})`}</div>
      </div>
    );
  }
}

ToolButton.propTypes = {
  buttonClass: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  iconClass: PropTypes.string.isRequired,
  updateCurrentTool: PropTypes.func.isRequired,
  shortcut: PropTypes.string.isRequired,
};

export default ToolButton;
