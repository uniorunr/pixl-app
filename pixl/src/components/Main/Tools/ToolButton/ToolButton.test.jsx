import React from 'react';
import renderer from 'react-test-renderer';
import ToolButton from './ToolButton';
import store from '../../../../stores/mainStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ToolButton
        buttonClass=""
        iconClass=""
        id=""
        shortcut=""
        updateCurrentTool={() => {}}
        store={store}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
