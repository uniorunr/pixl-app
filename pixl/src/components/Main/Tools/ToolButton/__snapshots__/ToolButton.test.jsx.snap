// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tools__container"
>
  <button
    className=""
    data-listener="true"
    id=""
    onClick={[Function]}
    type="button"
  >
    <i
      className=""
    />
  </button>
  <div
    className="tool__tooltip"
  >
     ()
  </div>
</div>
`;
